### PR TITLE
DRFT-131 Adds AUDIT log level to app logger

### PR DESCRIPTION
**Pull request by petrblaho**
Use it as `logger.audit("message")`
to see output as `[2021-02-18 15:29:43 +0000] [29] [AUDIT] message`.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [x] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by petrblaho** [comment]
https://issues.redhat.com/browse/DRFT-131

**Review by CodeHeeler — APPROVED** [review]
LGTM, thanks for handling this!

**Comment by petrblaho** [comment]
@CodeHeeler thank you

**Review by eherget — CHANGES_REQUESTED** [review]
I'm concerned we're not checking that keys "identity", "user" and "username" exist. If the x-rh-identity header ever changes or currently allows some of these to not exist, it will cause our code to crash.  Let's make sure we don't crash our code because of KeyError exceptions.

File: kerlescan/audit_logging.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+AUDIT_LEVEL_NUM = int((logging.INFO + logging.WARNING) / 2)
+
+
+def audit(self, message, *args, **kws):
+    if self.isEnabledFor(AUDIT_LEVEL_NUM):
+        request = kws.pop("request", None)
+
+        status = None
+        status = "success" if kws.pop("success", False) else None
+        status = "failure" if kws.pop("failure", False) else None
```
Comment (eherget) [inline kerlescan/audit_logging.py:17]: This requires the caller to provide success=True or failure=True.  Would it make more sense for the caller to provide just success=[True|False] and then set status = "success" or "failure" based on the value True or False?

File: kerlescan/audit_logging.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import json
+import base64
+
+from kerlescan.service_interface import get_key_from_headers
+
+AUDIT_LEVEL_NAME = "AUDIT"
+AUDIT_LEVEL_NUM = int((logging.INFO + logging.WARNING) / 2)
+
+
+def audit(self, message, *args, **kws):
```
Comment (eherget) [inline kerlescan/audit_logging.py:11]: Can we add a comment describing what kws args that can be passed and their expected values?

**Review by eherget — CHANGES_REQUESTED** [review]

**Comment by petrblaho** [comment]
I hope I addressed your review notes, @eherget

**Review by CodeHeeler — APPROVED** [review]
Thanks, I think this addresses all the concerns that were raised.

**Review by eherget — APPROVED** [review]

**Comment by eherget** [comment]
Looks good now, thank you!!!

**Pull request merged**